scripts/complete_liquidity_provision.js

```diff
@@ -105,15 +116,29 @@ module.exports = async (callback) => {
       callback("Error: Choose a smaller fleetSize, otherwise your payload will be to big for Infura nodes")
     }
 
-    console.log(`2. Deploying ${argv.fleetSize} trading brackets`)
-    const bracketAddresses = await deployFleetOfSafes(masterSafe.address, argv.fleetSize, true)
-    console.log("List of bracket traders in one line:", bracketAddresses.join())
+    let bracketAddresses
+    if (argv.brackets.length > 0) {
+      console.log("==> Skipping safe deployment and using brackets safeOwners")
+      bracketAddresses = argv.brackets
+      // Ensure that safes are all owned solely by masterSafe
+      for (const safeAddr of argv.brackets) {
+        const safeOwners = await (await GnosisSafe.at(safeAddr)).getOwners()
+        // TODO: write function like safeOwnedSolelyBy(safe, owner) -> Bool
+        if (!(safeOwners.length === 1 && safeOwners.includes(masterSafe.address))) {
+          callback(`Error: Bracket ${safeAddr} is not owned (or at least not solely) by master safe ${masterSafe.address}`)
+        }
+      }
+    } else {
+      console.log(`==> Deploying ${argv.fleetSize} trading brackets`)
+      bracketAddresses = await deployFleetOfSafes(masterSafe.address, argv.fleetSize, true)
+      console.log("List of bracket traders in one line:", bracketAddresses.join())
+    }
 
     // Sleeping for 3 seconds to make sure Infura nodes have processed all newly deployed contracts so that
     // they can be awaited.
     await sleep(3000)
```

Comment:
The sleep statement can go into the else branch of above

Reply:
Yes good catch! I had actually meant to do this earlier.